homeassistant/components/garmin_connect/sensor.py

```diff
@@ -109,7 +103,21 @@ def icon(self):
     @property
     def state(self):
         """Return the state of the sensor."""
-        return self._state
+        if not self.coordinator.data or not self.coordinator.data[self._type]:
+            return None
+
+        value = self.coordinator.data[self._type]
+        if "Duration" in self._type or "Seconds" in self._type:
+            value = value // 60
+        elif "Mass" in self._type or self._type == "weight":
+            value = value / 1000
+        elif self._type == "nextAlarm":
+            active_alarms = calculate_next_active_alarms(
+                self.coordinator.data[self._type]
+            )
+            if active_alarms:
+                self._state = active_alarms[0]
+        return round(value, 2)
```

Comment:
Should we always round?

Reply:
No, all but the time values, implemented check on device_class